### PR TITLE
Add convenience method for Value#toMap(keyMapper, v -> v)

### PR DESCRIPTION
**Pull request by MrcJkb**

### CODE REVIEW

File: src/main/java/io/vavr/Value.java [hunk 1]
```diff
@@ -1053,6 +1053,18 @@ default CharSeq toCharSeq() {
...
+    /**
+     * Converts this to a {@link Map}, with the elements as values.
+     *
+     * @param keyMapper   A function that maps an element to a key
+     * @param <K>         The key type
+     * @return A new {@link HashMap}.
+     */
+    default <K> Map<K, T> toMapBy(Function<? super T, ? extends K> keyMapper) {
+        Objects.requireNonNull(keyMapper, "keyMapper is null");
+        return toMap(keyMapper, v -> v);
```
Comment (frecco75) [inline src/main/java/io/vavr/Value.java:1065]: identity() instead of v -> v

**Review by frecco75 — APPROVED** [review]

**Comment by danieldietrich** [comment]
Hi & thank you for your suggestion. We would also need additional methods for `toSortedMap`, `toLinkedMap` and maybe for `toJavaMap`. I think the cost/income ratio isn't worth it to add such an amount of additional methods.

**Review by danieldietrich — COMMENTED** [review]
Hi & thank you for your suggestion. We would also need additional methods for toSortedMap, toLinkedMap and maybe for toJavaMap. I think the cost/income ratio isn't worth it to add such an amount of additional methods.

**Comment by danieldietrich** [comment]
I will close this as discussed.

**Pull request closed without merging**